java/src/main/java/org/rocksdb/NativeLibraryLoader.java

```diff
@@ -75,37 +75,43 @@ public synchronized void loadLibrary(final String tmpDir) throws IOException {
   void loadLibraryFromJar(final String tmpDir)
       throws IOException {
     if (!initialized) {
-      final File temp;
-      if (tmpDir == null || tmpDir.equals("")) {
-        temp = File.createTempFile(tempFilePrefix, tempFileSuffix);
-      } else {
-        temp = new File(tmpDir, jniLibraryFileName);
-        if (!temp.createNewFile()) {
-          throw new RuntimeException("File: " + temp.getAbsolutePath()
-              + " could not be created.");
-        }
-      }
+      System.load(loadLibraryFromJarToTemp(tmpDir).getAbsolutePath());
+      initialized = true;
+    }
+  }
 
-      if (!temp.exists()) {
-        throw new RuntimeException("File " + temp.getAbsolutePath() + " does not exist.");
-      } else {
-        temp.deleteOnExit();
+  File loadLibraryFromJarToTemp(final String tmpDir)
+          throws IOException {
+    final File temp;
+    if (tmpDir == null || tmpDir.equals("")) {
```

Comment:
You could consider `tmpDir.isEmpty()` instead of `tmpDir.equals("")` as RocksJava is targeting Java 7 (but only using those features that are likely Android compatible).